### ISSUE TITLE
System.Buffers Documentations

### ISSUE THREAD

**Issue by Abdullah2993** [dotnet/corefx#14702]
Where can I find the System.Buffers Documentation?

**Comment by deepumi** [comment]
@Abdullah2993 not sure you have already checked this
[https://github.com/dotnet/corefx/issues/4547](https://github.com/dotnet/corefx/issues/4547)
[https://apisof.net/catalog/System.Buffers](https://apisof.net/catalog/System.Buffers)

**Comment by Abdullah2993** [comment]
@deepumi I mean real documentation. Not available [in docs](https://docs.microsoft.com)

**Comment by karelz** [comment]
@alexperovich @safern isn't Buffers fairly new library? Maybe we just forgot to add it into API Reference docs?

**Comment by alexperovich** [comment]
Yes System.Buffers is a new library. I don't think we added it to the docs.

**Comment by Abdullah2993** [comment]
@alexperovich will it be added soon?

**Comment by karelz** [comment]
I don't think we will create the docs anytime soon - mainly we are blocked on capability to add new API reference docs. The story around API reference docs should be finalized during spring.
If there is huge need (i.e. multiple customers asking for it), we might think about some temporary workarounds where to place docs.

If you have specific questions, or the API surface is not clear on its own, ask here and we will try to help you.

**Comment by pmlyon** [comment]
@karelz, is it allowed to resize arrays retrieved from ArrayPool before returning them? I don't see it mentioned in the docs that are available in source: https://github.com/dotnet/corefx/blob/master/src/System.Buffers/src/System/Buffers/ArrayPool.cs

**Comment by safern** [comment]
AFAIK you have to rent a new piece of memory to the new size and copy the elements from the old and then return the old one.

**Comment by karelz** [comment]
There is no real resize on arrays anyway, [`Array.Resize`](https://github.com/dotnet/coreclr/blob/v2.0.3/src/mscorlib/src/System/Array.cs#L48-L69) will just allocate new array. Resizing capability is not really available in .NET.
